script/build.js

```diff
@@ -136,6 +137,28 @@ smith.use(ignore(ignoreList));
 // plugin chain.
 smith.use(filenames());
 
+smith.use((files, metalsmith, done) => {
+  runningCount = 0;
+  Object.keys(files).forEach((filename) => {
+    if (filename.match(/\.md$/) !== null) {
+      const fullPath = "content/pages/" + filename;
+      if (fs.existsSync(fullPath)) {
+        runningCount += 1;
+        git.log({ file: fullPath }, function(err, log) {
+          if (log.all.length > 1 && files[filename]) {
+            files[filename].lastupdate = log.latest.date;
+          }
+        }).then(function() {
+          runningCount -= 1;
+          if (runningCount == 0) {
+            done();
+          }
+        });
+      }
+    }
+  });
+});
+
```

Comment:
if these were all promises you could do https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all but unless you are going to instantiate a promise and return for each then stuff then into the array of promises and wrap in all... I think this is the simplest.